transcend/data_source_sombra.go

```diff
@@ -0,0 +1,73 @@
+package transcend
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	graphql "github.com/hasura/go-graphql-client"
+	"github.com/transcend-io/terraform-provider-transcend/transcend/types"
+)
+
+func dataSourceSombra() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceSombraRead,
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"url": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The url to lookup a sombra instance by",
+			},
+		},
+	}
+}
+
+func dataSourceSombraRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
```

Comment:
If I remember correctly, `interface{}` is like the `Any` of Go. I'm wondering if we can use a more specific type here?

Reply:
@MadDataScience We do it this to avoid having to wrestle with the Hashicorp Terraform SDK on types, and just use reflection downstream to get the right type. It's a bit gnarly, for sure, but that's the pattern we've been following in the rest of the repo